### DISCUSSION TITLE
ArchLinux packages

### DISCUSSION THREAD

**Discussion by davispuh** [JanssenProject/jans#8957] [Show and tell]
Hi,

I have packaged Janssen Project in [AUR](https://aur.archlinux.org/packages/jans-auth-server) (Arch Linux User Repository)
* Jans Auth Server
* Jans Config API
* Jans SCIM
* Jans Text UI ("TUI")
* Jans FIDO
* Jans LDAP Link
* Jans SAML
* Jans Casa

Documentation available at https://wiki.archlinux.org/title/Jans

Also unfortunately I wasn't able to reuse existing configuration scripts so I had to write my own:
* See [generate_config.sh](https://aur.archlinux.org/cgit/aur.git/tree/generate_config.sh?h=jans)
* See [scopes.py](https://aur.archlinux.org/cgit/aur.git/tree/scopes.py?h=jans)
* See [scripts.py](https://aur.archlinux.org/cgit/aur.git/tree/scripts.py?h=jans)

Would be great if upstream config scripts were more flexible and reusable.

**Comment by ossdhaval** [comment]
Thanks @davispuh 

cc: @moabu @nynymike

**Comment by nynymike** [comment]
That's great. Will it stay in sync with Jans releases on Github?

**Reply by davispuh** [reply]
Not automatic, someone has to bump version in PKBUILD but that's trivial.
It will look like this
```diff
diff --git a/PKGBUILD b/PKGBUILD
index 206eb72..b667124 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -16,7 +16,7 @@ pkgname=(jans-config
          jans-link
 #         jans-lock
          jans-casa)
-pkgver=1.1.3
+pkgver=1.1.4
 pkgrel=1
 pkgdesc="Janssen Project - An open source digital identity platforms that scales. Java OAuth Authorization Server and a certified OpenID Connect Provider."
 arch=("any")
```

**Comment by moabu** [comment]
Thanks @davispuh !

Can you open an issue with your requests on the upstream scripts so we can possibly help make your job easier ?

**Reply by davispuh** [reply]
Sure, I created #9188. But currently since these scripts are already written everything works. They only need to be updated if new incompatible changes come up. And also to support more databases since currently I implemented only LDAP support.

**Reply by nynymike** [reply]
FYI, we are removing LDAP support very soon. The best database is Postgres.  There is no good open source LDAP server and our enterprise customers prefer cloud RDBMS to host the IDP.

**Comment by ossdhaval** [comment]
We will track the corresponding issue. Closing this discussion.
